### PR TITLE
Move REST codestarts to be catalog dependant (RESTEasy/Reactive, Spring Web)

### PR DESCRIPTION
**Pull request by ia3andy**
We keep them (duplicated) in the base for a few versions to keep backward compat.

### CODE REVIEW

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java [hunk 1]
```diff
@@ -120,17 +122,22 @@ private static List<ResourceLoader> getCodestartResourceLoaders(String baseCodes
         if (catalog != null) {
             // Load codestarts from catalog codestart artifacts
             final List<String> catalogCodestartArtifacts = getCodestartArtifacts(catalog);
-            for (String artifactCoords : catalogCodestartArtifacts) {
-                if (codestartsArtifacts.containsKey(artifactCoords)) {
-                    continue;
+            for (String coords : catalogCodestartArtifacts) {
+                if (codestartsArtifacts.containsKey(coords)) {
+                    // Make sure it overrides the previous codestarts
+                    codestartsArtifacts.remove(coords);
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java:128]: Looks like this if and remove could have been removed completely.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java [hunk 2]
```diff
@@ -120,17 +122,22 @@ private static List<ResourceLoader> getCodestartResourceLoaders(String baseCodes
...
 
         // Load codestarts from the given artifacts
-        for (String codestartArtifactCoords : extraCodestartsArtifactCoords) {
-            codestartsArtifacts.put(codestartArtifactCoords, DependencyUtils.toArtifact(codestartArtifactCoords));
+        for (String coords : extraCodestartsArtifactCoords) {
+            if (codestartsArtifacts.containsKey(coords)) {
+                // Make sure it overrides the previous codestarts
+                codestartsArtifacts.remove(coords);
+            }
+            codestartsArtifacts.put(coords, DependencyUtils.toArtifact(coords));
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java:140]: Same here.

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java:140]: @aloubyansky It's a LinkedHashMap, `put` is keeping the order if already present, it needs to be removed to have the order overridden when putting again.

Comment (aloubyansky) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java:140]: I see, thanks.

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**